lib/cylc/network/connection_validator.py

```diff
@@ -123,8 +162,66 @@ def mungeIdent(self, ident):
         (user, host, prog name).
 
         """
+        hash_ = self._get_hash()
         uuid, passphrase = ident
         prog_name = os.path.basename(sys.argv[0])
         if passphrase is None:
             passphrase = NO_PASSPHRASE
-        return (user, host, str(uuid), prog_name, md5(passphrase).digest())
+        return (user, host, str(uuid), prog_name, hash_(passphrase).digest())
+
+    def _compare_hmacs(self, hmac1, hmac2):
+        """Compare hmacs as securely as possible."""
+        try:
+            return hmac.compare_hmacs(hmac1, hmac2)
+        except AttributeError:
+            # < Python 2.7.7.
+            return (hmac1 == hmac2)
+
+    def _get_default_hash_name(self):
+        if hasattr(self, "_default_hash_name"):
+            return self._default_hash_name
+        return GLOBAL_CFG.get()['authentication']['hashes'][0]
+
+    def _get_hash(self, hash_name=None):
+        try:
+            return self.HASHES[hash_name]
+        except KeyError:
+            pass
+
+        hash_name_dest = hash_name
+        if hash_name is None:
+            hash_name = self._get_default_hash_name()
+
+        try:
+            import hashlib
+            self.HASHES[hash_name_dest] = getattr(hashlib, hash_name)
+        except ImportError:
+            # < Python 2.5
```

Comment:
We don't need to do this - cylc requires Python 2.6 or later.

http://cylc.github.io/cylc/#requirements